carma_guidance_plugins/include/carma_guidance_plugins/plugin_base_node.hpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2022 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#pragma once
+
+#include <gtest/gtest_prod.h>
+#include <rclcpp/rclcpp.hpp>
+#include <carma_planning_msgs/msg/plugin.hpp>
+#include <carma_wm_ros2/WMListener.hpp>
+#include <carma_wm_ros2/WorldModel.hpp>
+
+#include <carma_ros2_utils/carma_lifecycle_node.hpp>
+
+namespace carma_guidance_plugins
+{
+
+  /**
+   * \brief PluginBaseNode provides default functionality for all carma guidance plugins.
+   *        This includes basic state machine management (largely delegated to ROS2 lifecycle behavior), required interfaces, and plugin discovery
+   * 
+   * Extending classes must implement the on_configure_plugin method to load parameters, and my override the other state transitions methods on_<state>_plugin if desired.
+   * Additionally, extending classes must implement the methods such as get_name() which are used to populate the plugin discovery message.
+   * 
+   */
+  class PluginBaseNode : public carma_ros2_utils::CarmaLifecycleNode
```

Comment:
Is a system_alert handler built into the base class being extended here?

Reply:
CarmaLifecycleNode does not contain a system alert handler since it delegates that behavior to the lifecycle services. It does have a system alert publisher. 

Did you have a particular usage you were thinking of?

Reply:
Do the lifecycle services then handle the shutdown when system_alert FATAL is published? I can't recall from that base class if that was included, or if that's baked into to each subsystem management node.

Reply:
The subsytem controllers check for the FATAL and then send the lifecycle service calls based on their configured required nodes.